Mathlib/Geometry/Synthetic/Avigad/Axioms.lean

```diff
@@ -0,0 +1,237 @@
+/--Relations between sidedness and betweeness-/
+sameside12_of_B123_sameside13 : ∀ {a b c L}, B a b c → SameSide a c L → SameSide a b L
+/--Relations between sidedness and betweeness-/
+sameside_of_B_not_online_2 : ∀ {a b c L}, B a b c → OnLine a L → ¬OnLine b L → SameSide b c L
+/--Relations between sidedness and betweeness-/
+not_sameside13_of_B123_online2 : ∀ {a b c L}, B a b c → OnLine b L → ¬SameSide a c L
+/--Relations between sidedness and betweeness-/
+B_of_online_inter : ∀ {a b c L M}, a ≠ b → b ≠ c → a ≠ c → L ≠ M →
+  OnLine a L → OnLine b L → OnLine c L → OnLine b M → ¬SameSide a c M → B a b c
+/--Deducing sidedness from three lines intersecting-/
+not_sameside_of_sameside_sameside : ∀ {a b c d L M N}, OnLine a L → OnLine a M → OnLine a N →
+  OnLine b L → OnLine c M → OnLine d N → SameSide c d L → SameSide b c N → ¬SameSide b d M
+/--Deducing sidedness from three lines intersecting-/
+sameside_of_sameside_not_sameside : ∀ {a b c d L M N}, a ≠ b → OnLine a L → OnLine a M →
+  OnLine a N → OnLine b L → OnLine c M → OnLine d N → ¬OnLine d M → SameSide c d L →
+  ¬SameSide b d M → SameSide b c N
+/--Points of intersection of a line and circle-/
+B_of_linecircleinter : ∀ {a b c L α}, b ≠ c → OnLine a L →
+  OnLine b L → OnLine c L → OnCircle b α → OnCircle c α → InCircle a α → B b a c
+/--The points of intersection of circles are on opposite sides the line that joins the centers-/
+not_sameside_of_circle_inter : ∀ {a b c d L α β},  c ≠ d → α ≠ β →  OnLine a L → OnLine b L →
+  OnCircle c α → OnCircle c β → OnCircle d α → OnCircle d β → CenterCircle a α → CenterCircle b β →
+  CirclesInter α β → ¬SameSide c d L
+/--Condition for lines intersecting-/
+lines_inter_of_not_sameside : ∀ {a b L M}, OnLine a M → OnLine b M → ¬SameSide a b L →
+  LinesInter L M
+/--Condition for line circle intersection-/
+linecircleinter_of_not_sameside : ∀ {a b L α}, ¬SameSide a b L → OnCircle a α ∨ InCircle a α →
+  OnCircle b α ∨ InCircle b α → LineCircleInter L α
+/--Condition for line circle intersection-/
+linecircleinter_of_inside_online : ∀ {a L α}, OnLine a L → InCircle a α → LineCircleInter L α
+/--Condition for circle circle intersection-/
+circlesinter_of_inside_on_circle : ∀ {a b α β}, OnCircle b α → OnCircle a β → InCircle a α →
+  InCircle b β → CirclesInter α β
+/--A length is zero iff the points are equal-/
+length_eq_zero_iff : ∀ {a b}, length a b = 0 ↔ a = b
+/--Length is symmetric-/
+length_symm : ∀ a b, length a b = length b a
+/--Angles are symmetric across the middle point-/
+angle_symm : ∀ a b c, angle a b c = angle c b a
+/--Angles are nonnegative-/
+angle_nonneg : ∀ a b c, 0 ≤ angle a b c
+/--Lengths are nonnegative-/
+length_nonneg : ∀ a b, 0 ≤ length a b
+/--Areas are nonnegative-/
+area_nonneg : ∀ a b c, 0 ≤ area a b c
+/--Degenerate areas are zero-/
+degenerate_area : ∀ a b, area a a b = 0
+/--Area is completely symmetric-/
+area_invariant : ∀ a b c, area a b c = area c a b ∧ area a b c = area a c b
```

Comment:
I think usually we would break this up into two separate conditions

Reply:
Ok, I have done so now!